packages/terra-list/src/List.scss

```diff
@@ -1,16 +1,62 @@
-@import '~terra-mixins';
 @import './variables';
 
-.terra-List {
-  list-style: none;
-  margin: 0;
-  padding: 0;
-}
+// List Component
+// ========================================================
+/* stylelint-disable selector-class-pattern, selector-max-compound-selectors */
+:local {
+  .list {
+    list-style: none;
+    margin: 0;
+    padding: 0;
+  }
+
+  .divided > .list-item {
+    border-top: solid 1px $terra-list-divider-color;
```

Comment:
I think this would be close to what you are looking for in the [variables.css](https://github.com/cerner/terra-core/blob/master/packages/terra-badge/src/_variables.scss) and [list.scss](https://github.com/cerner/terra-core/blob/master/packages/terra-badge/src/Badge.scss) files:

variables.scss
```
:root {
--terra-list-border-width: 1px;
--terra-list-divider-color: ;
}

$terra-list-border-top: (
  border-top: solid var(--terra-list-border-width) var(--terra-list-divider-color);
);
$terra-list-border-bottom: (
   border-bottom: solid var(--terra-list-border-width) var(--terra-list-divider-color);
);
```
and in List.scss
```
.divided > .list-item {
    @include $terra-list-border-top;

    &:last-of-type {
      @include $terra-list-border-bottom;
    }
  }
```